### PR TITLE
Python script to upload sample data to S3

### PR DESCRIPTION
**Pull request by ryansteakley**
Python script to upload training and validation data usable for training and hyper-parameter tuning jobs.

### CODE REVIEW

File: samples/hyperparameter_tuning/my-hyperparameter-job.yaml [hunk 1]
```diff
@@ -63,9 +63,9 @@ spec:
         s3DataSource:
           s3DataType: S3Prefix
           # The source of the training data
-          s3URI: s3://<YOUR BUCKET/PATH>
+          s3URI: s3://<YOUR BUCKET/sagemaker/xgboost/train
```
Comment (mbaijal) [inline samples/hyperparameter_tuning/my-hyperparameter-job.yaml:66]: missing closing `>` bracket

File: samples/hyperparameter_tuning/my-hyperparameter-job.yaml [hunk 2]
```diff
@@ -74,9 +74,9 @@ spec:
         s3DataSource:
           s3DataType: S3Prefix
           # The source of the validation data
-          s3URI: s3://<YOUR BUCKET/PATH>
+          s3URI: s3://<YOUR BUCKET/sagemaker/xgboost/validation
```
Comment (mbaijal) [inline samples/hyperparameter_tuning/my-hyperparameter-job.yaml:77]: same comment for `>`

Comment (mbaijal) [reply samples/hyperparameter_tuning/my-hyperparameter-job.yaml:77]: check all files for this.

File: samples/training/s3_sample_data.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+## Download dataset from pinned commit
+url = "https://github.com/aws/amazon-sagemaker-examples/raw/af6667bd0be3c9cdec23fecda7f0be6d0e3fa3ea/sagemaker-debugger/xgboost_realtime_analysis/data_utils.py"
+urllib.request.urlretrieve(url, "data_utils.py")
+
+from data_utils import load_mnist, upload_to_s3
+
+prefix = "sagemaker/xgboost"
+train_file, validation_file = load_mnist()
+upload_to_s3(train_file, bucket, f"{prefix}/train/mnist.train.libsvm")
+upload_to_s3(validation_file, bucket, f"{prefix}/validation/mnist.validation.libsvm")
```
Comment (mbaijal) [inline samples/training/s3_sample_data.py:21]: can you run `black` on this file  if you have not already ? I think black adds a newline at the EOF.

**Comment by surajkota** [comment]
Remove downloaded binary file

File: samples/training/s3_sample_data.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import urllib.request
+from urllib.parse import urlparse
+
+###################################################################
+# This is the only thing that you need to change to run this code
+# Give the name of your S3 bucket
```
Comment (mbaijal) [inline samples/training/s3_sample_data.py:6]: nit: To make this a standalone script, this should probably have some comments about region and credentials. wdyt ?

File: samples/hyperparameter_tuning/my-hyperparameter-job.yaml (hunk 2 above)
Comment (ryansteakley) [reply samples/hyperparameter_tuning/my-hyperparameter-job.yaml:77]: noted should only exist for these files changed as I updated the path will change

File: samples/training/README.md [hunk 5]
```diff
@@ -6,6 +6,8 @@ This sample demonstrates how to start training jobs using your own training scri
 
 This sample assumes that you have completed the [common prerequisties](/samples/README.md).
 
+You will need training data uploaded to an S3 bucket. Run the [s3_sample_data.py](/samples/training/s3_sample_data.py) to upload sample data.
```
Comment (surajkota) [inline samples/training/README.md:9]: Write the command to run the script and say you replace bucket

**Review by surajkota — APPROVED** [review]

File: samples/training/s3_sample_data.py (hunk 4 above)
Comment (ryansteakley) [reply samples/training/s3_sample_data.py]: The script doesn't ask for their region only uploads to their s3 bucket. Is it necessary to remind user to pass a bucket in the region they are currently using as default?

**Review by surajkota — APPROVED** [review]

File: samples/training/s3_sample_data.py (hunk 4 above)
Comment (mbaijal) [reply samples/training/s3_sample_data.py]: ya, you could mention that.

File: samples/README.md [hunk 6]
```diff
@@ -20,7 +21,25 @@ SAGEMAKER_EXECUTION_ROLE_ARN=$(aws iam get-role --role-name ${SAGEMAKER_EXECUTIO
...
 echo $SAGEMAKER_EXECUTION_ROLE_ARN
 ```
-Note down the execution role ARN to use in samples
+Note down the execution role ARN to use in samples.
+
+### S3 Bucket
+Run the following commands to create a S3 bucket which is used by SageMaker service to access and upload data.
+
+```
+export S3_BUCKET_NAME="ack-bucket-$CLUSTER_NAME"
```
Comment (surajkota) [inline samples/README.md:30]: Add this: Use the region you used during installation unless you are trying cross region. SageMaker resources will be created in this region

```
export AWS_DEFAULT_REGION=<REGION>
export S3_BUCKET_NAME="ack-data-bucket-$AWS_DEFAULT_REGION"
```

File: samples/training/s3_sample_data.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+# Gets your bucket name from command line
+# Make sure you have AWS credentials and are in the proper AWS region for your bucket
+try:
+    bucket = str(sys.argv[1])
+except Exception as error:
+    print("Please pass your bucket name as a commandline argument")
+    sys.exit(1)
+
+# Download dataset from pinned commit
+url = "https://github.com/aws/amazon-sagemaker-examples/raw/af6667bd0be3c9cdec23fecda7f0be6d0e3fa3ea/sagemaker-debugger/xgboost_realtime_analysis/data_utils.py"
```
Comment (surajkota) [inline samples/training/s3_sample_data.py:14]: run [black tool](https://pypi.org/project/black/) on this script

File: samples/hyperparameter_tuning/README.md [hunk 8]
```diff
@@ -4,7 +4,14 @@ This sample demonstrates how to start hyperparameter jobs using your own hyperpa
 
 ## Prerequisites
 
-This sample assumes that you have completed the [common prerequisties](/samples/README.md).
+This sample assumes that you have completed the [common prerequisites](/samples/README.md).
+
+### Upload S3 Data
+
+You will need training data uploaded to an S3 bucket. Make sure you have AWS credentials and are in the proper AWS region for your bucket. Run the following python script replacing YOUR-BUCKET-NAME with the name of your S3 bucket.
```
Comment (surajkota) [inline samples/hyperparameter_tuning/README.md:11]: are in the ~~proper AWS region for your bucket~~

and have the bucket in the same region where you plan to create SageMaker resources


---

let's just use
python3 ../training/s3_sample_data.py `$S3_BUCKET_NAME` so the user doesn't need to replace. We assume the user has followed prereqs. If they haven't, the script will anyway throw an error anyway

**Review by surajkota — CHANGES_REQUESTED** [review]

**Review by surajkota — APPROVED** [review]
/approve
/lgtm

**Review by mbaijal — APPROVED** [review]

**Pull request merged**